In [2]:
import preprocessing as pp
import load_files as lf
import nlp_algorithms as nlp

import pandas as pd
import numpy as np

from sentence_transformers import SentenceTransformer
pd.set_option('display.max_colwidth', -1)

C:\Users\lucas\AppData\Local\Temp/ipykernel_8692/2826523860.py:9: FutureWarning: Passing a negative integer is deprecated in version 1.0 and will not be supported in future version. Instead, use None to not limit the column width.
  pd.set_option('display.max_colwidth', -1)


# Checando dados train v0.1 e v0.2

In [3]:
train_v1 = pd.read_csv('dados/train v0.1.csv')

train_v1_enen = train_v1[(train_v1['url1_lang'] == 'en') & (train_v1['url2_lang'] == 'en')]
train_v1_enen.head()

url1_lang url2_lang                pair_id  \
0  en        en        1484084337_1484110209   
1  en        en        1484396422_1483924666   
2  en        en        1484698254_1483758694   
3  en        en        1576314516_1576455088   
4  en        en        1484036253_1483894099   

                                                                                                                                                       link1  \
0  https://www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
1  https://www.stlucianewsonline.com/guyana-three-injured-after-car-crashes-into-utility-pole/                                                                 
2  https://www.teaparty.org/trump-brings-in-2020-at-mar-a-lago-were-going-to-have-a-great-year-423052/                                                         
3  https://gadgets.ndtv.com/apps/news/zomato-uber-eats-business-acquisition-india-all-stock-deal-2167155                                                       
4  https://news.yahoo.com/india-approves-third-moon-mission-085759387.html                                                                                     

                                                                                                                                                                      link2  \
0  https://www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-11ea-bffe-020c88b3f120_story.html   
1  https://www.thestar.com/news/world/europe/2020/01/01/fire-kills-animals-at-zoo-in-western-germany.html                                                                     
2  https://www.timesofisrael.com/trump-says-he-does-not-expect-war-with-iran-likes-peace/                                                                                     
3  https://gadgets.ndtv.com/internet/news/indian-online-food-delivery-market-to-hit-usd-8-billion-by-2020-google-bcg-report-2171043                                           
4  https://www.channelnewsasia.com/news/asia/india-targets-new-moon-mission-in-2020-12225344                                                                                  

                                                                                                                                                                        ia_link1  \
0  https://web.archive.org/web/www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
1  https://web.archive.org/web/www.stlucianewsonline.com/guyana-three-injured-after-car-crashes-into-utility-pole/                                                                 
2  https://web.archive.org/web/www.teaparty.org/trump-brings-in-2020-at-mar-a-lago-were-going-to-have-a-great-year-423052/                                                         
3  https://web.archive.org/web/gadgets.ndtv.com/apps/news/zomato-uber-eats-business-acquisition-india-all-stock-deal-2167155                                                       
4  https://web.archive.org/web/news.yahoo.com/india-approves-third-moon-mission-085759387.html                                                                                     

                                                                                                                                                                                       ia_link2  \
0  https://web.archive.org/web/www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-11ea-bffe-020c88b3f120_story.html   
1  https://web.archive.org/web/www.thestar.com/news/world/europe/2020/01/01/fire-kills-animals-at-zoo-in-western-germany.html                                                                     
2  https://web.archive.org/web/www.

In [4]:
train_v1[(train_v1['url1_lang'] == 'en') & (train_v1['url2_lang'] == 'en')].shape

(1761, 14)

In [5]:
train_v1[(train_v1['url1_lang'] == 'de') & (train_v1['url2_lang'] == 'de')].shape

(414, 14)

In [6]:
train_v1[(train_v1['url1_lang'] == 'es') & (train_v1['url2_lang'] == 'es')].shape

(444, 14)

In [10]:
train_v1_de = train_v1[(train_v1['url1_lang'] == 'de') & (train_v1['url2_lang'] == 'de')]
train_v1_de.head()

url1_lang url2_lang                pair_id  \
1622  de        de        1483969308_1483894885   
1623  de        de        1483980389_1483874154   
1624  de        de        1484160744_1484028776   
1625  de        de        1483994737_1484672934   
1626  de        de        1484167939_1483914365   

                                                                                                                                                    link1  \
1622  https://www.hna.de/kultur/tv-kino/marco-cerullo-dschungelcamp-2020-kandidat-rtl-bachelor-in-paradise-zr-13359743.html?cmp=defrss                      
1623  https://www.onetz.de/oberpfalz/neutras-etzelwang/silberdistel-neutras-ehrt-jahresmeister-id2934113.html                                               
1624  http://zsz.ch/wirtschaft/unternehmen-und-konjunktur/Das-grosse-Raetsel-um-die-Flucht-des-Autozaren/story/18590310                                     
1625  https://www.braunschweiger-zeitung.de/helmstedt/lehre-velpke-grasleben/article228034567/Zum-Jahresende-legen-die-Velpker-noch-mal-richtig-los.html    
1626  https://www.luzernerzeitung.ch/zentralschweiz/luzern/9-personen-festgenommen-viel-arbeit-fuer-die-luzerner-polizei-in-der-silvesternacht-ld.1181932   

                                                                                                                     link2  \
1622  https://www.spiegel.de/kultur/tv/dschungelcamp-bei-rtl-ex-minister-guenther-krause-macht-mit-a-1303279.html#ref=rss    
1623  https://www.onetz.de/oberpfalz/burkhardsreuth-trabitz/froehliche-weise-ernster-botschaft-id2933913.html                
1624  https://deutsch.rt.com/der-nahe-osten/96336-ueberraschende-flucht-aus-japan-ex-renault-boss-carlos-ghosn-im-libanon/   
1625  https://www.noz.de/lokales/melle/artikel/1968021/ein-meisterkonzert-der-leiseren-toene-in-melle                        
1626  http://szo.de/?arid=11165129                                                                                           

                                                                                                                                                                     ia_link1  \
1622  https://web.archive.org/web/www.hna.de/kultur/tv-kino/marco-cerullo-dschungelcamp-2020-kandidat-rtl-bachelor-in-paradise-zr-13359743.html?cmp=defrss                      
1623  https://web.archive.org/web/www.onetz.de/oberpfalz/neutras-etzelwang/silberdistel-neutras-ehrt-jahresmeister-id2934113.html                                               
1624  https://web.archive.org/web/zsz.ch/wirtschaft/unternehmen-und-konjunktur/Das-grosse-Raetsel-um-die-Flucht-des-Autozaren/story/18590310                                    
1625  https://web.archive.org/web/www.braunschweiger-zeitung.de/helmstedt/lehre-velpke-grasleben/article228034567/Zum-Jahresende-legen-die-Velpker-noch-mal-richtig-los.html    
1626  https://web.archive.org/web/www.luzernerzeitung.ch/zentralschweiz/luzern/9-personen-festgenommen-viel-arbeit-fuer-die-luzerner-polizei-in-der-silvesternacht-ld.1181932   

                                                                                                                                      ia_link2  \
1622  https://web.archive.org/web/www.spiegel.de/kultur/tv/dschungelcamp-bei-rtl-ex-minister-guenther-krause-macht-mit-a-1303279.html#ref=rss    
1623  https://web.archive.org/web/www.onetz.de/oberpfalz/burkhardsreuth-trabitz/froehliche-weise-ernster-botschaft-id2933913.html                
1624  https://web.archive.org/web/deutsch.rt.com/der-nahe-osten/96336-ueberraschende-flucht-aus-japan-ex-renault-boss-carlos-ghosn-im-libanon/   
1625  https://web.archive.org/web/www.noz.de/lokales/melle/artikel/1968021/ein-meisterkonzert-der-leiseren-toene-in-melle                        
1626  https://web.archive.org/web/szo.de/?arid=11165129                                                                                          

      Geography  Entities  Time  Narrative   Overall     S

In [15]:
train_v1_es = train_v1[(train_v1['url1_lang'] == 'es') & (train_v1['url2_lang'] == 'es')]
train_v1_es.head()

url1_lang url2_lang                pair_id  \
1859  es        es        1484225684_1484168941   
1860  es        es        1485001551_1483980929   
1861  es        es        1483981600_1484104403   
1862  es        es        1484222388_1484383793   
1863  es        es        1484112831_1484051337   

                                                                                                                   link1  \
1859  http://www.poresto.net/2020/01/01/se-realizan-compras-de-ultima-hora-para-fin-de-ano/                                
1860  http://www.diarioelmarino.cl/2020/01/01/realizaran-obras-de-mejoramiento-en-el-bosque-municipal-de-pichilemu/        
1861  https://www.lagacetadesalamanca.es/internacional/estos-son-los-paises-mas-peligrosos-para-viajar-en-2020-GC1985855   
1862  https://www.elliberal.com.ar/noticia/516082/nino-sufrio-quemaduras-luego-poner-canita-voladora-bidon-nafta           
1863  http://diariotextual.com/inicio/index.php/2020/01/01/francisco-pidio-disculpas-incidente-una-fiel-plaza-san-pedro/   

                                                                                                                                                  link2  \
1859  http://chaconoticias.com.ar/heridos-de-pirotecnia/                                                                                                  
1860  http://www.soychile.cl/Arica/Sociedad/2020/01/01/632560/Arica-Con-ambiente-festivo-se-inauguro-la-ultima-plaza-del-ano-en-poblacion-Tarapaca.aspx   
1861  http://www.telesurtv.net/news/condena-fip-49-asesinatos-periodistas-2019-20200101-0013.html                                                         
1862  https://elinformante.mx/?p=45432                                                                                                                    
1863  http://www.clarin.com/mundo/papa-denuncia-violencia-mujeres-pornografia_0_TSLQfUpO.html                                                             

                                                                                                                                     ia_link1  \
1859  https://web.archive.org/web/www.poresto.net/2020/01/01/se-realizan-compras-de-ultima-hora-para-fin-de-ano/                                
1860  https://web.archive.org/web/www.diarioelmarino.cl/2020/01/01/realizaran-obras-de-mejoramiento-en-el-bosque-municipal-de-pichilemu/        
1861  https://web.archive.org/web/www.lagacetadesalamanca.es/internacional/estos-son-los-paises-mas-peligrosos-para-viajar-en-2020-GC1985855    
1862  https://web.archive.org/web/www.elliberal.com.ar/noticia/516082/nino-sufrio-quemaduras-luego-poner-canita-voladora-bidon-nafta            
1863  https://web.archive.org/web/diariotextual.com/inicio/index.php/2020/01/01/francisco-pidio-disculpas-incidente-una-fiel-plaza-san-pedro/   

                                                                                                                                                                    ia_link2  \
1859  https://web.archive.org/web/chaconoticias.com.ar/heridos-de-pirotecnia/                                                                                                  
1860  https://web.archive.org/web/www.soychile.cl/Arica/Sociedad/2020/01/01/632560/Arica-Con-ambiente-festivo-se-inauguro-la-ultima-plaza-del-ano-en-poblacion-Tarapaca.aspx   
1861  https://web.archive.org/web/www.telesurtv.net/news/condena-fip-49-asesinatos-periodistas-2019-20200101-0013.html                                                         
1862  https://web.archive.org/web/elinformante.mx/?p=45432                                                                                                                     
1863  https://web.archive.org/web/www.clarin.com/mundo/papa-denuncia-violencia-mujeres-pornografia_0_TSLQfUpO.html                                                             

      Geography  Entities  Time  Narrative  Overall  Style  Tone  
1859  4.0        4.0       

In [19]:
data_path = 'dados/train v0.1/'

lista_docs = []
lista_error = []
lista_vazio = []
values = train_v1_enen[['pair_id', 'Overall']]
#values = train_v1_de[['pair_id', 'Overall']]
#values = train_v1_es[['pair_id', 'Overall']]

for index, values in values.iterrows():
    
    try:
        
        json_pair = lf.get_json_document_pair(data_path, values['pair_id'])
        text_doc1 = json_pair[0]['text']
        text_doc2 = json_pair[1]['text']
        
        if ( len(text_doc1) > 0 ) and ( len(text_doc2) > 0 ):
            lista_docs.append((values['pair_id'], text_doc1, text_doc2, values['Overall']))
        else:
            lista_vazio.append(values['pair_id'])
    
    except:
        lista_error.append(values['pair_id'])

In [20]:
df_text = pd.DataFrame(lista_docs,  columns=['pair_id', 'doc1', 'doc2', 'Overall'])
#df_text.head(1)

In [21]:
df_text.shape

(1357, 4)

valores finais:

train v0.1:
- en-en: 1357
- de-de: 264
- es-es: 354

train v0.2:
- en-en: 1610 (1577 duplicados)
- de-de: 260 (tudo duplicado)
- es-es: 299 (261 duplicado)

checar se tem duplicados (os do v0.1 estao no v0.2? SIM! :( ))

ou seja: considerar apenas o v0.2

In [22]:
train_v2 = pd.read_csv('dados/train v0.2.csv')

train_v2_en = train_v2[(train_v2['url1_lang'] == 'en') & (train_v1['url2_lang'] == 'en')]
train_v2_en.head()

url1_lang url2_lang                pair_id  \
0  en        en        1484084337_1484110209   
1  en        en        1484396422_1483924666   
2  en        en        1484698254_1483758694   
3  en        en        1576314516_1576455088   
4  en        en        1484036253_1483894099   

                                                                                                                                                       link1  \
0  https://www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
1  https://www.stlucianewsonline.com/guyana-three-injured-after-car-crashes-into-utility-pole/                                                                 
2  https://www.teaparty.org/trump-brings-in-2020-at-mar-a-lago-were-going-to-have-a-great-year-423052/                                                         
3  https://gadgets.ndtv.com/apps/news/zomato-uber-eats-business-acquisition-india-all-stock-deal-2167155                                                       
4  https://news.yahoo.com/india-approves-third-moon-mission-085759387.html                                                                                     

                                                                                                                                                                      link2  \
0  https://www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-11ea-bffe-020c88b3f120_story.html   
1  https://www.thestar.com/news/world/europe/2020/01/01/fire-kills-animals-at-zoo-in-western-germany.html                                                                     
2  https://www.timesofisrael.com/trump-says-he-does-not-expect-war-with-iran-likes-peace/                                                                                     
3  https://gadgets.ndtv.com/internet/news/indian-online-food-delivery-market-to-hit-usd-8-billion-by-2020-google-bcg-report-2171043                                           
4  https://www.channelnewsasia.com/news/asia/india-targets-new-moon-mission-in-2020-12225344                                                                                  

                                                                                                                                                                        ia_link1  \
0  https://web.archive.org/web/www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
1  https://web.archive.org/web/www.stlucianewsonline.com/guyana-three-injured-after-car-crashes-into-utility-pole/                                                                 
2  https://web.archive.org/web/www.teaparty.org/trump-brings-in-2020-at-mar-a-lago-were-going-to-have-a-great-year-423052/                                                         
3  https://web.archive.org/web/gadgets.ndtv.com/apps/news/zomato-uber-eats-business-acquisition-india-all-stock-deal-2167155                                                       
4  https://web.archive.org/web/news.yahoo.com/india-approves-third-moon-mission-085759387.html                                                                                     

                                                                                                                                                                                       ia_link2  \
0  https://web.archive.org/web/www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-11ea-bffe-020c88b3f120_story.html   
1  https://web.archive.org/web/www.thestar.com/news/world/europe/2020/01/01/fire-kills-animals-at-zoo-in-western-germany.html                                                                     
2  https://web.archive.org/web/www.

In [23]:
train_v2_es = train_v2[(train_v2['url1_lang'] == 'es') & (train_v1['url2_lang'] == 'es')]
train_v2_es.head()

url1_lang url2_lang                pair_id  \
1859  es        es        1484112820_1484058586   
1860  es        es        1484690350_1484607381   
1861  es        es        1484116465_1484108871   
1862  es        es        1484130702_1484158388   
1863  es        es        1484690337_1484075524   

                                                                                                                         link1  \
1859  http://diario99.com.ar/enjambre-de-moscas-el-dolor-intelectual-de-flori-melnechuk/                                         
1860  https://www.diariocordoba.com/noticias/cordobalocal/ayuntamiento-convoca-13-plazas-oficial-policia-local_1345528.html      
1861  https://mundo.sputniknews.com/america-latina/202001011090031209-maduro-felicita-a-cuba-por-los-61-anos-de-la-revolucion/   
1862  https://www.yucatan.com.mx/mexico/presuntos-sicarios-abren-fuego-contra-vehiculo-en-felipe-carrillo-puerto                 
1863  https://www.diariocordoba.com/noticias/cultura/zarzuela-strauss-abrir-ano_1345503.html                                     

                                                                                                                                        link2  \
1859  https://www.minutoneuquen.com/nacionales/2020/1/1/sabina-frederic-sostendra-el-operativo-patricia-bullrich-en-las-fronteras-176807.html   
1860  http://www.boe.es/diario_boe/txt.php?id=BOE-A-2020-39                                                                                     
1861  https://primicia.com.ve/nacion/venezuela-felicita-a-cuba-por-61-aniversario-de-la-revolucion/                                             
1862  https://www.yucatan.com.mx/yucatan/una-veladora-encendida-causa-un-incendio-en-un-condominio-de-uman                                      
1863  https://sevilla.abc.es/andalucia/cordoba/sevi-agenda-cordoba-concierto-nuevo-orquesta-cordoba-202001011808_noticia.html                   

                                                                                                                                          ia_link1  \
1859  https://web.archive.org/web/diario99.com.ar/enjambre-de-moscas-el-dolor-intelectual-de-flori-melnechuk/                                        
1860  https://web.archive.org/web/www.diariocordoba.com/noticias/cordobalocal/ayuntamiento-convoca-13-plazas-oficial-policia-local_1345528.html      
1861  https://web.archive.org/web/mundo.sputniknews.com/america-latina/202001011090031209-maduro-felicita-a-cuba-por-los-61-anos-de-la-revolucion/   
1862  https://web.archive.org/web/www.yucatan.com.mx/mexico/presuntos-sicarios-abren-fuego-contra-vehiculo-en-felipe-carrillo-puerto                 
1863  https://web.archive.org/web/www.diariocordoba.com/noticias/cultura/zarzuela-strauss-abrir-ano_1345503.html                                     

                                                                                                                                                         ia_link2  \
1859  https://web.archive.org/web/www.minutoneuquen.com/nacionales/2020/1/1/sabina-frederic-sostendra-el-operativo-patricia-bullrich-en-las-fronteras-176807.html   
1860  https://web.archive.org/web/www.boe.es/diario_boe/txt.php?id=BOE-A-2020-39                                                                                    
1861  https://web.archive.org/web/primicia.com.ve/nacion/venezuela-felicita-a-cuba-por-61-aniversario-de-la-revolucion/                                             
1862  https://web.archive.org/web/www.yucatan.com.mx/yucatan/una-veladora-encendida-causa-un-incendio-en-un-condominio-de-uman                                      
1863  https://web.archive.org/web/sevilla.abc.es/andalucia/cordoba/sevi-agenda-cordoba-concierto-nuevo-orquesta-cordoba-202001011808_noticia.html                   

      Geography  Entities  Time  Narrative  Overall  Style  Tone  
1859  2.5        4.0       1.5   4.0        4.0      4.0    2.5   
1860  3.0        2.0

In [24]:
train_v2_de = train_v2[(train_v2['url1_lang'] == 'de') & (train_v1['url2_lang'] == 'de')]
train_v2_de.head()

url1_lang url2_lang                pair_id  \
1622  de        de        1483978591_1484232740   
1623  de        de        1484159213_1484536109   
1624  de        de        1484167909_1484472101   
1625  de        de        1484217119_1484002781   
1626  de        de        1483972874_1483852995   

                                                                                                                                            link1  \
1622  https://www.presseportal.de/blaulicht/pm/50510/4481401                                                                                        
1623  https://www.weser-kurier.de/deutschland-welt/deutschland-welt-politik_artikel,-ein-stresstest-fuer-die-foederale-familie-_arid,1886602.html   
1624  https://www.luzernerzeitung.ch/zentralschweiz/das-wurde-aus-den-neujahrsbabys-von-damals-ld.1181965                                           
1625  https://www.tagblatt.ch/international/aufdringliche-frau-bekommt-vom-papst-auf-die-finger-ld.1181955                                          
1626  https://www.mittelbayerische.de/region/neumarkt-nachrichten/rueckenwind-fuer-johann-karg-21102-art1865134.html                                

                                                                                                                   link2  \
1622  https://www.dzonline.de/NRW/4082602-Braende-Rakete-fliegt-durch-gekipptes-Fenster?utm_source=RSS                     
1623  https://burgenland.orf.at/stories/3028263/                                                                           
1624  https://www.freiepresse.de/vogtland/auerbach/die-vogtlaendischen-neujahrsbabys-2020-mila-und-lasse-artikel10693332   
1625  https://www.journalistenwatch.com/2020/01/01/ohne-beherrschung-papst/                                                
1626  https://steiermark.orf.at/stories/3028150/                                                                           

                                                                                                                                                             ia_link1  \
1622  https://web.archive.org/web/www.presseportal.de/blaulicht/pm/50510/4481401                                                                                        
1623  https://web.archive.org/web/www.weser-kurier.de/deutschland-welt/deutschland-welt-politik_artikel,-ein-stresstest-fuer-die-foederale-familie-_arid,1886602.html   
1624  https://web.archive.org/web/www.luzernerzeitung.ch/zentralschweiz/das-wurde-aus-den-neujahrsbabys-von-damals-ld.1181965                                           
1625  https://web.archive.org/web/www.tagblatt.ch/international/aufdringliche-frau-bekommt-vom-papst-auf-die-finger-ld.1181955                                          
1626  https://web.archive.org/web/www.mittelbayerische.de/region/neumarkt-nachrichten/rueckenwind-fuer-johann-karg-21102-art1865134.html                                

                                                                                                                                    ia_link2  \
1622  https://web.archive.org/web/www.dzonline.de/NRW/4082602-Braende-Rakete-fliegt-durch-gekipptes-Fenster?utm_source=RSS                     
1623  https://web.archive.org/web/burgenland.orf.at/stories/3028263/                                                                           
1624  https://web.archive.org/web/www.freiepresse.de/vogtland/auerbach/die-vogtlaendischen-neujahrsbabys-2020-mila-und-lasse-artikel10693332   
1625  https://web.archive.org/web/www.journalistenwatch.com/2020/01/01/ohne-beherrschung-papst/                                                
1626  https://web.archive.org/web/steiermark.orf.at/stories/3028150/                                                                           

      Geography  Entities      Time  Narrative   Overall     Style      Tone  
1622  1.000000   2.333333  1.000000  2.333333   2.333333  1.666667  1.000000  
1623  3.666667   4.0

In [29]:
data_path = 'dados/train v0.2/'

lista_docs = []
lista_error = []
lista_vazio = []
#values = train_v2_en[['pair_id', 'Overall']]
#values = train_v2_de[['pair_id', 'Overall']]
values = train_v2_es[['pair_id', 'Overall']]

for index, values in values.iterrows():
    
    try:
        
        json_pair = lf.get_json_document_pair(data_path, values['pair_id'])
        text_doc1 = json_pair[0]['text']
        text_doc2 = json_pair[1]['text']
        
        if ( len(text_doc1) > 0 ) and ( len(text_doc2) > 0 ):
            lista_docs.append((values['pair_id'], text_doc1, text_doc2, values['Overall']))
        else:
            lista_vazio.append(values['pair_id'])
    
    except:
        lista_error.append(values['pair_id'])

In [30]:
df_text = pd.DataFrame(lista_docs,  columns=['pair_id', 'doc1', 'doc2', 'Overall'])
df_text.shape

(299, 4)

-----------------------------

In [35]:
train_en = pd.concat([train_v1_enen, train_v2_en])

In [39]:
train_en[train_en.duplicated()].shape

(1577, 14)

In [40]:
train_es = pd.concat([train_v1_es, train_v2_es])

In [41]:
train_es[train_es.duplicated()].shape

(310, 14)

In [42]:
train_de = pd.concat([train_v1_de, train_v2_de])

In [43]:
train_de[train_de.duplicated()].shape

(261, 14)

# Checando dados evaluation

In [45]:
eval_nolabel = pd.read_csv('dados/eval/evaluation_nolabel.csv')
eval_nolabel.head()

url1_lang url2_lang                pair_id  \
0  en        en        1484189203_1484121193   
1  en        en        1484011097_1484011106   
2  en        en        1484039488_1484261803   
3  en        en        1484332324_1484796748   
4  en        en        1484012256_1484419682   

                                                                                                                                  link1  \
0  https://wsvn.com/news/local/broward/police-2-men-stole-tools-from-lowes-in-davie/                                                      
1  https://www.zdnet.com/article/autoclerk-database-leaked-customer-government-and-military-personal-records/                             
2  https://www.presstelegram.com/2019/12/31/ducks-are-own-worst-enemies-in-sloppy-loss-in-las-vegas/                                      
3  https://www.financialexpress.com/india-news/west-bengal-republic-day-tableau-rejected-by-centre-mamata-banerjee-latest-news/1810097/   
4  https://www.birminghammail.co.uk/whats-on/music-nightlife-news/bars-clubs-you-loved-lost-17492049                                      

                                                                                                                                                  link2  \
0  https://wsvn.com/news/local/no-swim-advisory-lifted-for-deerfield-beach-pier/                                                                          
1  https://securityboulevard.com/2019/10/best-westerns-massive-data-leak-179gb-amazon-database-open-to-all/                                               
2  https://boingboing.net/2020/01/01/woody-guthries-1943-new-year.html                                                                                    
3  https://www.news18.com/news/india/not-the-first-congress-did-7-times-bjp-reminds-as-row-over-bengal-maharashtra-r-day-tableau-continues-2443643.html   
4  http://m.fightbacknews.org/2020/1/1/top-20-films-2010s                                                                                                 

                                                                                                                                                           ia_link1  \
0  https://web.archive.org/web/https://wsvn.com/news/local/broward/police-2-men-stole-tools-from-lowes-in-davie/                                                      
1  https://web.archive.org/web/https://www.zdnet.com/article/autoclerk-database-leaked-customer-government-and-military-personal-records/                             
2  https://web.archive.org/web/https://www.presstelegram.com/2019/12/31/ducks-are-own-worst-enemies-in-sloppy-loss-in-las-vegas/                                      
3  https://web.archive.org/web/https://www.financialexpress.com/india-news/west-bengal-republic-day-tableau-rejected-by-centre-mamata-banerjee-latest-news/1810097/   
4  https://web.archive.org/web/https://www.birminghammail.co.uk/whats-on/music-nightlife-news/bars-clubs-you-loved-lost-17492049                                      

                                                                                                                                                                           ia_link2  
0  https://web.archive.org/web/https://wsvn.com/news/local/no-swim-advisory-lifted-for-deerfield-beach-pier/                                                                         
1  https://web.archive.org/web/https://securityboulevard.com/2019/10/best-westerns-massive-data-leak-179gb-amazon-database-open-to-all/                                              
2  https://web.archive.org/web/https://boingboing.net/2020/01/01/woody-guthries-1943-new-year.html                                                                                   
3  https://web.archive.org/web/https://www.news18.com/news/india/not-the-first-congress-did-7-times-bjp-reminds-as-row-over-bengal-maharashtra-r-day-tableau-continues-2443643.html  
4  https://web.archive.org/web/htt

In [46]:
eval = pd.read_csv('dados/eval/final_evaluation_data.csv')
eval.head()

url1_lang url2_lang                pair_id  \
0  en        en        1484189203_1484121193   
1  en        en        1484011097_1484011106   
2  en        en        1484039488_1484261803   
3  en        en        1484332324_1484796748   
4  en        en        1484012256_1484419682   

                                                                                                                                  link1  \
0  https://wsvn.com/news/local/broward/police-2-men-stole-tools-from-lowes-in-davie/                                                      
1  https://www.zdnet.com/article/autoclerk-database-leaked-customer-government-and-military-personal-records/                             
2  https://www.presstelegram.com/2019/12/31/ducks-are-own-worst-enemies-in-sloppy-loss-in-las-vegas/                                      
3  https://www.financialexpress.com/india-news/west-bengal-republic-day-tableau-rejected-by-centre-mamata-banerjee-latest-news/1810097/   
4  https://www.birminghammail.co.uk/whats-on/music-nightlife-news/bars-clubs-you-loved-lost-17492049                                      

                                                                                                                                                  link2  \
0  https://wsvn.com/news/local/no-swim-advisory-lifted-for-deerfield-beach-pier/                                                                          
1  https://securityboulevard.com/2019/10/best-westerns-massive-data-leak-179gb-amazon-database-open-to-all/                                               
2  https://boingboing.net/2020/01/01/woody-guthries-1943-new-year.html                                                                                    
3  https://www.news18.com/news/india/not-the-first-congress-did-7-times-bjp-reminds-as-row-over-bengal-maharashtra-r-day-tableau-continues-2443643.html   
4  http://m.fightbacknews.org/2020/1/1/top-20-films-2010s                                                                                                 

                                                                                                                                                           ia_link1  \
0  https://web.archive.org/web/https://wsvn.com/news/local/broward/police-2-men-stole-tools-from-lowes-in-davie/                                                      
1  https://web.archive.org/web/https://www.zdnet.com/article/autoclerk-database-leaked-customer-government-and-military-personal-records/                             
2  https://web.archive.org/web/https://www.presstelegram.com/2019/12/31/ducks-are-own-worst-enemies-in-sloppy-loss-in-las-vegas/                                      
3  https://web.archive.org/web/https://www.financialexpress.com/india-news/west-bengal-republic-day-tableau-rejected-by-centre-mamata-banerjee-latest-news/1810097/   
4  https://web.archive.org/web/https://www.birminghammail.co.uk/whats-on/music-nightlife-news/bars-clubs-you-loved-lost-17492049                                      

                                                                                                                                                                           ia_link2  \
0  https://web.archive.org/web/https://wsvn.com/news/local/no-swim-advisory-lifted-for-deerfield-beach-pier/                                                                          
1  https://web.archive.org/web/https://securityboulevard.com/2019/10/best-westerns-massive-data-leak-179gb-amazon-database-open-to-all/                                               
2  https://web.archive.org/web/https://boingboing.net/2020/01/01/woody-guthries-1943-new-year.html                                                                                    
3  https://web.archive.org/web/https://www.news18.com/news/india/not-the-first-congress-did-7-times-bjp-reminds-as-row-over-bengal-maharashtra-r-day-tableau-continues-2443643.html   
4  https://web.archive.org/we

# Per annotator data

In [47]:
anot_train = pd.read_csv('dados/per_annotator_training_data.csv')
anot_train.head()

url1_lang url2_lang                pair_id  \
0  en        en        1484084337_1484110209   
1  en        en        1484084337_1484110209   
2  en        en        1484084337_1484110209   
3  en        en        1484396422_1483924666   
4  en        en        1484396422_1483924666   

                                                                                                                                                       link1  \
0  https://www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
1  https://www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
2  https://www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
3  https://www.stlucianewsonline.com/guyana-three-injured-after-car-crashes-into-utility-pole/                                                                 
4  https://www.stlucianewsonline.com/guyana-three-injured-after-car-crashes-into-utility-pole/                                                                 

                                                                                                                                                                      link2  \
0  https://www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-11ea-bffe-020c88b3f120_story.html   
1  https://www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-11ea-bffe-020c88b3f120_story.html   
2  https://www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-11ea-bffe-020c88b3f120_story.html   
3  https://www.thestar.com/news/world/europe/2020/01/01/fire-kills-animals-at-zoo-in-western-germany.html                                                                     
4  https://www.thestar.com/news/world/europe/2020/01/01/fire-kills-animals-at-zoo-in-western-germany.html                                                                     

                                                                                                                                                                                ia_link1  \
0  https://web.archive.org/web/https://www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
1  https://web.archive.org/web/https://www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
2  https://web.archive.org/web/https://www.washingtonpost.com/local/virginia-man-arrested-in-fatal-dui-crash-in-west-virginia/2020/01/01/740fbc7a-2cbe-11ea-bffe-020c88b3f120_story.html   
3  https://web.archive.org/web/https://www.stlucianewsonline.com/guyana-three-injured-after-car-crashes-into-utility-pole/                                                                 
4  https://web.archive.org/web/https://www.stlucianewsonline.com/guyana-three-injured-after-car-crashes-into-utility-pole/                                                                 

                                                                                                                                                                                               ia_link2  \
0  https://web.archive.org/web/https://www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-11ea-bffe-020c88b3f120_story.html   
1  https://web.archive.org/web/https://www.washingtonpost.com/world/the_americas/haitis-leader-marks-independence-day-amid-security-concerns/2020/01/01/dc4033a4-2cc5-

In [48]:
anot_eval = pd.read_csv('dados/per_annotator_evaluation_data.csv')
anot_eval.head()

url1_lang url2_lang                pair_id  \
0  en        en        1484189203_1484121193   
1  en        en        1484189203_1484121193   
2  en        en        1484011097_1484011106   
3  en        en        1484011097_1484011106   
4  en        en        1484039488_1484261803   

                                                                                                        link1  \
0  https://wsvn.com/news/local/broward/police-2-men-stole-tools-from-lowes-in-davie/                            
1  https://wsvn.com/news/local/broward/police-2-men-stole-tools-from-lowes-in-davie/                            
2  https://www.zdnet.com/article/autoclerk-database-leaked-customer-government-and-military-personal-records/   
3  https://www.zdnet.com/article/autoclerk-database-leaked-customer-government-and-military-personal-records/   
4  https://www.presstelegram.com/2019/12/31/ducks-are-own-worst-enemies-in-sloppy-loss-in-las-vegas/            

                                                                                                      link2  \
0  https://wsvn.com/news/local/no-swim-advisory-lifted-for-deerfield-beach-pier/                              
1  https://wsvn.com/news/local/no-swim-advisory-lifted-for-deerfield-beach-pier/                              
2  https://securityboulevard.com/2019/10/best-westerns-massive-data-leak-179gb-amazon-database-open-to-all/   
3  https://securityboulevard.com/2019/10/best-westerns-massive-data-leak-179gb-amazon-database-open-to-all/   
4  https://boingboing.net/2020/01/01/woody-guthries-1943-new-year.html                                        

                                                                                                                                 ia_link1  \
0  https://web.archive.org/web/https://wsvn.com/news/local/broward/police-2-men-stole-tools-from-lowes-in-davie/                            
1  https://web.archive.org/web/https://wsvn.com/news/local/broward/police-2-men-stole-tools-from-lowes-in-davie/                            
2  https://web.archive.org/web/https://www.zdnet.com/article/autoclerk-database-leaked-customer-government-and-military-personal-records/   
3  https://web.archive.org/web/https://www.zdnet.com/article/autoclerk-database-leaked-customer-government-and-military-personal-records/   
4  https://web.archive.org/web/https://www.presstelegram.com/2019/12/31/ducks-are-own-worst-enemies-in-sloppy-loss-in-las-vegas/            

                                                                                                                               ia_link2  \
0  https://web.archive.org/web/https://wsvn.com/news/local/no-swim-advisory-lifted-for-deerfield-beach-pier/                              
1  https://web.archive.org/web/https://wsvn.com/news/local/no-swim-advisory-lifted-for-deerfield-beach-pier/                              
2  https://web.archive.org/web/https://securityboulevard.com/2019/10/best-westerns-massive-data-leak-179gb-amazon-database-open-to-all/   
3  https://web.archive.org/web/https://securityboulevard.com/2019/10/best-westerns-massive-data-leak-179gb-amazon-database-open-to-all/   
4  https://web.archive.org/web/https://boingboing.net/2020/01/01/woody-guthries-1943-new-year.html                                        

               annotator_id  GEO  ENT  TIME  NAR  OVERALL  STYLE  TONE  
0  60d78ec10ab6b67496bb5992  1.0  4.0  3.0   4.0  3.0      1.0    1.0   
1  60d78e810ab6b6744561617e  2.0  4.0  1.0   4.0  4.0      1.0    2.0   
2  60d78ef60ab6b674e5ff0fc6  1.0  2.0  1.0   1.0  1.0      3.0    2.0   
3  60d78f210ab6b6751cf79baa  1.0  2.0  1.0   1.0  1.0      4.0    3.0   
4  60d32f0d0ab6b60191107859 NaN   4.0  4.0   4.0  4.0      4.0    4.0